### PR TITLE
Homework modern cpp #2 

### PR DESCRIPTION
**Pull request by mateusz-talma**
Done all tasks from README.md

### CODE REVIEW

**Review by MaciejSalamonski — APPROVED** [review]
Everything looks OK to me :) good job :)

**Review by WBigaj — APPROVED** [review]
I have checked your code and it looks good 👍

File: shapes/main.cpp [hunk 1]
```diff
@@ -1,96 +1,127 @@
...
+    }
     return (first->getArea() < second->getArea());
-}
+};
 
-bool perimeterBiggerThan20(shared_ptr<Shape> s)
-{
-    if(s)
-        return (s->getPerimeter() > 20);
+auto perimeterBiggerThan20(shared_ptr<Shape> shape) {
```
Comment (ziobron) [inline shapes/main.cpp:25]: it should be a lambda function

File: shapes/main.cpp [hunk 2]
```diff
@@ -1,96 +1,127 @@
...
+    AddShapeToCollection(shapes, make_shape<Circle>(3.0));
+    //AddShapeToCollection(shapes, nullptr); compile error
+    shapes.emplace_back(nullptr);
+    AddShapeToCollection(shapes, make_shape<Circle>(4.0));
+    AddShapeToCollection(shapes, make_shape<Rectangle>(10.0, 5.0));
+    AddShapeToCollection(shapes, make_shape<Square>(3.0));
+    AddShapeToCollection(shapes, make_shape<Square>(4.0));
+}
+
+void FillMapCollection(Collection& shapes, MapCollection& mapCollection) {
```
Comment (ziobron) [inline shapes/main.cpp:101]:
```suggestion
void FillMapCollection(const Collection& shapes, MapCollection& mapCollection) {
```
The first param should not be changed.

**Review by ziobron — APPROVED** [review]

**Comment by ziobron** [comment]
15 XP za Nowoczesny C++ SFINAE (4447)
5 XP za Nowoczesny C++ alignas (4448)
5 XP za Nowoczesny C++ delegating constructors (4449)
5 XP za Nowoczesny C++ lambda (4450)
5 XP za Nowoczesny C++ structured bindings (4451)
15 XP za Nowoczesny C++ variadic templates (4452)

{ "requirement_ids": [4447, 4448, 4449, 4450, 4451, 4452] }

**Pull request closed without merging**